public/js/main.js

```diff
@@ -31,13 +31,42 @@ const state = {
 async function initCamera() {
   state.video = document.getElementById('video');
 
-  const stream = await navigator.mediaDevices.getUserMedia({
-    video: {
-      facingMode: 'user',
-    },
-    audio: false,
-  });
-  state.video.srcObject = stream;
+  if (!navigator.mediaDevices || !navigator.mediaDevices.getUserMedia) {
+    // eslint-disable-next-line no-alert
+    return alert('Your browser does not support the webcam functionality');
+  }
+
+  /* navigator.mediaDevices.getUserMedia().then(
+    (m) => console.log('Media stream=', m),
+    (err) => console.log('Media stream err=', err) // Gets called
+  ); */
+
+  navigator.mediaDevices.ondevicechange = (evt) => {
+    // eslint-disable-next-line no-console
+    console.log('Device change detected:', evt);
+    // Do some updates here
```

Comment:
Similarly here?

Reply:
Yeah, I was initially thinking of having it so that when the webcam is freed, a handler like that or `stream.onactive` could catch it and resume/restart the process but none of those approaches worked so I'll remove that.